src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8083,7 +8083,7 @@ void reportBadDelegateParameter(BindingDiagnosticBag bag, MethodSymbol sourceInv
 
             TypeWithState resultType = calculateResultType(targetTypeWithNullability, fromExplicitCast, resultState, isSuppressed, targetType);
 
-            if (operandType.Type?.IsErrorType() != true && !targetType.IsErrorType())
+            if (operandType.Type?.IsErrorType() != true && !targetType.IsErrorType() && !conversionOperand.HasErrors)
```

Comment:
I was curious if `operandType.Type?.IsErrorType != true` could be outright replaced with `!conversionOperand.HasErrors`. It feels like if the operandType is an error type, then the bound expression which has that type should have errors. Do you know?